celery/backends/redis.py

```diff
@@ -102,9 +103,11 @@ def on_state_change(self, meta, message):
         self._maybe_cancel_ready_task(meta)
 
     def start(self, initial_task_id, **kwargs):
-        self._pubsub = self.backend.client.pubsub(
-            ignore_subscribe_messages=True,
-        )
+        if self._pubsub is None:
```

Comment:
Shouldn't we do this inside the property getter instead?

Reply:
Setting this property does not work this way...I forked off this branch and this method was not the solution. The _pubsub property is delegated to the current self._thread property. See below because there is a typo in the property getter @sihrc

Reply:
Would you mind contributing the fix?

Reply:
I won't be able to push to @sihrc's branch unless I have permission on his fork unfortunately. And I don't think opening a new PR from my fork is a great idea. Hoping Chris will see this in the next couple days.

Reply:
I think there should be one redis client and one connection_pool, we only need different pubsub objects in different threads.